api/src/test/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleEntitlementCalculatorTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimantTestDataFactory.aValidClaimant;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.aValidVoucherEntitlement;
+
+@ExtendWith(MockitoExtension.class)
+class CycleEntitlementCalculatorTest {
+
+    private static final int PAYMENT_CYCLE_DURATION_IN_DAYS = 3;
+    private static final int NUMBER_OF_CALCULATION_PERIODS = 3;
+
+    private EntitlementCalculator entitlementCalculator = mock(EntitlementCalculator.class);
+
+    private CycleEntitlementCalculator cycleEntitlementCalculator
+            = new CycleEntitlementCalculator(PAYMENT_CYCLE_DURATION_IN_DAYS, NUMBER_OF_CALCULATION_PERIODS, entitlementCalculator);
+
+    @Test
+    void shouldThrowExceptionWhenDurationIsNotDivisibleByNumberOfCalculationPeriods() {
+        // 10 is not divisible by 3, so should throw an exception
+        IllegalArgumentException thrown = catchThrowableOfType(
+                () -> new CycleEntitlementCalculator(10, 3, null), IllegalArgumentException.class);
+
+        assertThat(thrown.getMessage()).isEqualTo("Payment cycle duration of 10 days is not divisible by number of calculation periods 3");
+    }
+
+    @Test
+    void shouldThrowExceptionWhenDurationIsZero() {
+        // 10 is not divisible by 3, so should throw an exception
+        IllegalArgumentException thrown = catchThrowableOfType(
+                () -> new CycleEntitlementCalculator(0, 1, null), IllegalArgumentException.class);
+
+        assertThat(thrown.getMessage()).isEqualTo("Payment cycle duration can not be zero");
+    }
+
+    @Test
+    void shouldThrowExceptionWhenNumberOrCalculationPeriodsIsZero() {
+        // 10 is not divisible by 3, so should throw an exception
+        IllegalArgumentException thrown = catchThrowableOfType(
+                () -> new CycleEntitlementCalculator(1, 0, null), IllegalArgumentException.class);
+
+        assertThat(thrown.getMessage()).isEqualTo("Number of calculation periods can not be zero");
+    }
+
+    @Test
+    void shouldCallEntitlementCalculatorForEachEntitlementDate() {
```

Comment:
This test is very confusing as the VoucherEntitlement has 2 children under 1 and 1 between 1 and 4 yet you only have 1 date of birth being added to the CycleEntitlementCalculator, could you line these up properly so it make sense? I also think you need to explain the calculations more as its really hard to figure out whether this test is correct or not.

Reply:
I can line them up but that might make things more confusing. Basically the number of children is irrelevant here as we're mocking the class that uses them.

Reply:
Yes, ok, in which case, for this one test could we create a voucher entitlement in this test that does match the dates of birth to remove the confusion and not have to change the test data factory?

Reply:
Done.